external/mcpclient/src/mcp-client-plugin.ts

```diff
@@ -189,8 +160,16 @@ export class McpClientPlugin implements ChatPromptPlugin<'mcpClient', McpClientP
     }
   }
 
-  private makeMcpClientPlugin(serverUrl: string) {
-    const transport = this.createTransport(serverUrl);
+  private async makeMcpClientPlugin(
+    serverUrl: string,
+    headers: ValueOrFactory<Record<string, string>> | undefined
+  ) {
+    let transport;
+    if (this.createTransport != null) {
```

Comment:
nit: should typically use !== for Comparisions. Not a huge deal tho.

Reply:
for null-checks it's ok (that's the only one where it makes sense. it checks for both undefined, or null)